minecraft/minecraft-worldscanner/src/test/java/minecraft/worldscanner/ScannerTest.java

```diff
@@ -186,6 +186,32 @@ public void findTileEntities() throws IOException {
         PorkUtil.unsafe_forceGC();
     }
 
+    @Test
+    public void getBiomes() throws IOException {
+        try (MinecraftSave save = this.getTestWorld()) {
+            new WorldScanner(save.world(0))
+                    .addProcessor(col -> {
+                        if ((col.getX() & 0x1F) == 31 && (col.getZ() & 0x1F) == 31) {
+                            System.out.printf("Scanning region (%d,%d)\n", col.getX() >> 5, col.getZ() >> 5);
+                        }
+                    })
+                    .addProcessor((current, estimatedTotal, chunk, access) -> {
+                        int x = chunk.getX() << 4;
+                        int z = chunk.getZ() << 4;
+                        for (int xx = 15; xx >= 0; xx--) {
+                            for (int zz = 15 - (xx & 1); zz >= 0; zz -= 2) {
```

Comment:
what's with the weird indexing on the Z axis? this will read biomes in a checkerboard pattern...

Reply:
oops, i copied that loop from the doublechest test